src/adera/Machines/Rocket.cpp

```diff
@@ -163,9 +170,86 @@ void SysMachineRocket::update_physics()
     }
 }
 
+void SysMachineRocket::attach_plume_effect(ActiveEnt ent)
+{
+    using Magnum::GL::Mesh;
+    using Magnum::GL::Texture2D;
+    using Magnum::Trade::ImageData2D;
+    using namespace Magnum::Math::Literals;
+
+    ActiveEnt plumeNode = entt::null;
+
+    auto findPlumeHandle = [&](ActiveEnt ent)
+    {
+        ACompHierarchy& node = m_scene.reg_get<ACompHierarchy>(ent);
+        if (node.m_name.compare(0, 9, "fx_plume_") == 0)
+        {
+            plumeNode = ent;
+            return false;  // terminate search
+        }
+        return true;
+    };
+
+    m_scene.hierarchy_traverse(ent, findPlumeHandle);
+
+    if (plumeNode == entt::null)
+    {
+        std::cout << "ERROR: could not find plume anchor for MachineRocket "
+            << static_cast<int>(ent) << "\n";
+        return;
+    }
+    std::cout << "MachineRocket "
+        << static_cast<int>(ent) << "'s associated plume: "
+        << static_cast<int>(plumeNode) << "\n";
+
+    // Get plume effect
+    Package& pkg = m_scene.get_application().debug_find_package("lzdb");
+    std::string plumeAnchorName = m_scene.reg_get<ACompHierarchy>(plumeNode).m_name;
+    std::string effectName = plumeAnchorName.substr(3, plumeAnchorName.length() - 3);
+    DependRes<PlumeEffectData> plumeEffect = pkg.get<PlumeEffectData>(effectName);
+    if (plumeEffect.empty())
+    {
+        std::cout << "ERROR: couldn't find plume effect " << effectName << "!\n";
+        return;
+    }
+
+    // Get plume mesh
+    Package& glResources = m_scene.get_context_resources();
+    DependRes<Mesh> plumeMesh = glResources.get<Mesh>(plumeEffect->meshName);
+    if (plumeMesh.empty())
+    {
+        plumeMesh = AssetImporter::compile_mesh(plumeEffect->meshName, pkg, glResources);
+    }
+
+    // Get plume tex (TEMPORARY: just grab noise1024.png)
+    const std::string texName = "OSPData/adera/noise1024.png";
```

Comment:
If I have time, I may generalize this for this commit, although there's no rush as there's only one plume effect right now